chore(dev): simplify dual-database local dev setup

webjunkie opened the pull request:
## Changes

Consolidate local dev from two Postgres containers to one container hosting multiple databases.

**Before:** Two containers (`db` on :5432, `persons_db` on :5434)  
**After:** One container on :5432 with three databases (`posthog`, `cyclotron`, `posthog_persons`)

## Why

Two containers were introduced to avoid global transaction ID conflicts in production ingestion code. Local dev doesn't run that code path, so physical separation isn't needed. This reduces complexity without affecting functionality.

Benefits:
- Simpler setup - one container instead of two
- No hard-coded :5434 port references
- Easier to add future databases (just add init script)

## What Changed

**Docker infrastructure:**
- Removed `persons_db` service from docker-compose files
- Removed `persons-db` volume definitions  
- Added `docker/postgres-init-scripts/create-persons-db.sh` (idempotent init script)
- Single Postgres container creates all databases on startup

**Configuration updates:**
- Django: `posthog/settings/data_stores.py` - updated persons database defaults
- Plugin-server: `plugin-server/src/config/config.ts` - all PERSONS_*_DATABASE_URL defaults
- Rust: `rust/bin/migrate-all` - updated default database URL
- Cymbal: `bin/start-rust-service` - updated PERSONS_URL for cymbal service

**CI workflows:**
- Updated `.github/workflows/ci-e2e-playwright.yml` - migration commands
- Updated `.github/workflows/ci-nodejs.yml` - PERSONS_*_DATABASE_URL env vars
- Updated `.github/workflows/ci-dagster.yml` - migration commands
- Updated `.github/workflows/ci-rust.yml` - persons database URLs
- Updated `.github/workflows/ci-backend.yml` - migration commands

**Local dev tooling:**
- Deleted `bin/check_persons_db_up` (redundant, use `check_postgres_up posthog_persons`)
- Updated `bin/mprocs.yaml` and `bin/mprocs-minimal.yaml` - use generic check pattern
- Updated `docker-compose.dev-minimal.yml` - removed persons_db service

All port references changed from `:5434` → `:5432`. Environment variables still override defaults for production/custom setups.

## Migration for Developers

Fresh start (recommended):
```bash
docker compose ... down -v  # removes old volumes
docker compose ... up       # creates new single-container setup
```

The init script will automatically create all three databases (`posthog`, `cyclotron`, `posthog_persons`) on first startup.

## Test Plan

Verified:
- [x] Single Postgres container starts successfully
- [x] All three databases created automatically (posthog, cyclotron, posthog_persons)
- [x] Generic check script works: `bin/check_postgres_up posthog_persons`
- [x] Demo data generation runs successfully against new setup
- [x] Full `bin/start` workflow
- [x] CI workflows pass

## Future Database Pattern

Adding new databases is now straightforward - see [comment](https://github.com/PostHog/posthog/pull/39302#issuecomment-3385082353) for the pattern.

webjunkie committed:
chore(docker): consolidate postgres into single container

Remove redundant persons_db service from docker-compose.dev.yml.
Create posthog_persons database via init script in main container.

Main container now hosts three databases:
- posthog (main)
- cyclotron
- posthog_persons

This matches production's logical separation without physical container
duplication, avoiding global transaction ID conflicts in local dev where
ingestion code doesn't run.

webjunkie committed:
chore(config): update database URLs for single postgres instance

Change all localhost:5434 references to localhost:5432 to use the
consolidated postgres container.

Updates:
- Django settings default URL
- Plugin-server dev/test defaults
- Rust migration script defaults
- Rust test configurations
- CI workflows (playwright, nodejs, dagster, rust, backend)
- Local dev tooling (mprocs.yaml)

All services now connect to port 5432 with appropriate database names.
Environment variables still override defaults for flexibility.

nickbest-ph reviewed (approved):
I tested running migrations and generating demo data/loading the application after I made the suggested change to `./bin/check_persons_db_up` and it successfully populated the persons database with data 👍 . Thank you for making this simpler and more lightweight

webjunkie committed:
chore(merge): resolve conflicts with master

Merge master and resolve conflicts in:
- ci-nodejs.yml: keep both BEHAVIORAL_COHORTS_DATABASE_URL and updated port 5432
- docker-compose.dev.yml: keep behavioral_cohorts_db, remove persons_db

Master introduced behavioral_cohorts_db on port 5433.
This PR consolidates persons database into main db container on port 5432.
Both changes are compatible.

webjunkie committed:
chore(merge): resolve conflicts after behavioral_cohorts revert

Behavioral cohorts feature was reverted in master.
Resolved conflicts to keep our port 5432 changes while removing behavioral cohorts references.

- Removed BEHAVIORAL_COHORTS_DATABASE_URL from ci-nodejs.yml
- Removed persons_db and behavioral_cohorts_db containers from docker-compose.dev.yml
- Main db container on port 5432 now hosts posthog, cyclotron, and posthog_persons databases

webjunkie committed:
chore(dev): remove redundant persons_db check script and consolidate minimal setup

Remove check_persons_db_up script since check_postgres_up already handles arbitrary database checks via parameter. Update all references to use the generic pattern instead.

Also clean up docker-compose.dev-minimal.yml to match the single-container setup and fix remaining port 5434 references in rust service config.
